kafka-rest/src/main/java/io/confluent/kafkarest/v2/KafkaConsumerState.java

```diff
@@ -374,6 +458,51 @@ boolean hasNextCached() {
     return consumerRecords.poll();
   }
 
+  /**
+   * A topic partition offset, and its associated timestamp.
+   */
+  public static final class OffsetTimestamp {
```

Comment:
given the existence of `org.apache.kafka.clients.consumer.OffsetAndTimestamp`, what's the value in having this class as well?

also, internally, I'd be inclined to leave timestamp as `long` for as long as possible (see another conversation about the association with time being a little bit arbitrary / user defined).